domain/network/service/interface.go

```diff
@@ -13,12 +13,19 @@ import (
 	"github.com/juju/juju/environs"
 )
 
-// Provider is the interface that the network service requires to be able to
+// ProviderWithNetworking is the interface that the network service requires to be able to
 // interact with the underlying provider.
-type Provider interface {
+type ProviderWithNetworking interface {
 	environs.Networking
 }
 
+// ProviderWithZones is the interface that the network service requires to be able to
```

Comment:
I don't think this comment is adding value here. Specifically something like....

`ProviderWithZones describes the interface needed from providers that support reporting the zones available for use`.

could be a better helper introduction.

Reply:
Sure, was just copying the existing comment from the previously landed work. Will update both.